### ISSUE TITLE
testing readings

### ISSUE THREAD

**Issue by huss** [OpenEnergyDashboard/OED#962]
### Is your feature request related to a problem? Please describe.

The[ Design Document on testing](https://github.com/OpenEnergyDashboard/DesignDocs/blob/main/testing/testing.md) specifies readings tests that are desired. Once a test is complete and merged it is labeled as done. We would like the others completed.

### Describe the solution you'd like

Implement the tests. This is an issue that is generally done by a new developer to get used to working with OED. They normally do 1 or 2 tests for this process and then move on to something else so there are tests for others.

### Additional context

See information in the design document on doing a test, You must allocate the test as described before beginning work on a test.

**Comment by aliceduongg** [comment]
Hi, I am working with this file (in Line Readings) with the description: "should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as kg of CO2"

**Comment by dzheng13341** [comment]
Hi, I will work on the file with the description: "should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as metric ton of CO2 & chained"

**Comment by aliceduongg** [comment]
Hi, I am working with this file (in Line Readings) with the description: "should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as lbs of CO2 & chained & reversed"

**Comment by mrathgeber** [comment]
I have completed test cases with the descriptions:

- partial days/hours for daily gives only full days (line reading)
- 13 day bars for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as kWh (bar reading)

**Comment by Dominier** [comment]
Hello, I am working with bar reading test case B14 with the description: "1 day bars for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as lbs of CO2 & chained & reversed"

**Comment by TamNganLy** [comment]
Hi, I am working on test B11: "1 day bars for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as BTU reverse conversion"

**Comment by ngocvbh** [comment]
Hello, I am working on line reading test L20: "should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as lbs of CO2 & chained & reversed"

**Comment by amancior23** [comment]
Hello my team is working on B13

**Comment by alexis0705** [comment]
Hello my team is working on test B8

**Comment by abccodes** [comment]
Hello, my team is working on test B12.

**Comment by yadhiramorales** [comment]
Hello, my team is working on test B6.

**Comment by abccodes** [comment]
Hello, my team is working on test L2.

**Comment by JaedonSpurlock01** [comment]
Hello, our team is working on test B10.

**Comment by huss** [comment]
> Hello, my team is working on test L2.

@abccodes I just noticed that L2 is already in the code and marked complete in the Design Docs. Another person indicated they are doing L20 so I'm uncertain what test you are working on.

**Comment by abccodes** [comment]
@huss sorry my mistake, I was assigned issue L2 through codeday and was unaware it was already completed. I will find another issue to work on and delete the previous message.

**Comment by abccodes** [comment]
> > Hello, my team is working on test L2.
> 
> @abccodes I just noticed that L2 is already in the code and marked complete in the Design Docs. Another person indicated they are doing L20 so I'm uncertain what test you are working on.

@huss sorry my mistake, I was assigned issue L2 through codeday and was unaware it was already completed. I will find another issue to work on and delete the previous message.

**Comment by abccodes** [comment]
Me and my team are working on test B9.

**Comment by alexis0705** [comment]
Hi, I am working on test L18

**Comment by HanCotterell** [comment]
Working on Test B5 implementation.

**Comment by huss** [comment]
> Working on Test B5 implementation.

@HanCotterell This test is in PR #1108. I'm hoping you know about that effort. If not, then please let me know as it should soon be merged and complete B5.

**Comment by HanCotterell** [comment]
@huss No, I didn’t know about that. I checked to make sure no one was doing it before commenting I was working on it a few weeks ago. I’ve finished the code and was planning to submit the PR today.

**Comment by huss** [comment]
> @huss No, I didn’t know about that. I checked to make sure no one was doing it before commenting I was working on it a few weeks ago. I’ve finished the code and was planning to submit the PR today.

Okay. Are you with CodeDay? I'm trying to determine how best to resolve this overlapping work. Since you seem to be the only one who claimed it, you probably will get priority unless there are other circumstances.

**Comment by HanCotterell** [comment]
Yes, I did do a different project with CodeDay but my team didn’t submit as a group so I didn’t get credit/my name on the project I worked on. That’s why I waited till after the projects were due/program was finished to check if this test was claimed before commenting and completing it.

**Comment by ShubhamPatilsd** [comment]
CodeDay students will be working on tests LG2-LG21, LG8, LG25, BG1-BG16

**Comment by alfarj83** [comment]
I'll be working on test B15.

**Comment by VivianBelle17** [comment]
I will be working on test LG7

**Comment by TuongPhamM** [comment]
Me and my team will be working on test LG19

**Comment by JasonKimy** [comment]
My team and I will work on LG5 units of energy

**Comment by Michael-Kowata** [comment]
My team and I will be working on test LG6.

**Comment by sumtzehern** [comment]
Me and my team will be working on test LG4

**Comment by GaryZLG** [comment]
Me and my team will be working on test LG20.

**Comment by adedralin** [comment]
We have done LG21 test. @ChelsyLan and @codingfiddler and Gyugang Lee.

**Comment by MangoPicante** [comment]
My team is working on test case BG2

**Comment by Ben4C** [comment]
My team is working on test case BG3

**Comment by ken-ni** [comment]
My team and I will be working on LG8.

**Comment by ianLewis8** [comment]
I am working on test BG4: "13 day bars for 15 + 20 minute reading intervals and quantity units with +-inf start/end time & kWh as kWh".

**Comment by nlin21** [comment]
I will be working on LG13

**Comment by danielshid** [comment]
I will be working on BG6

**Comment by williamw04** [comment]
I will be working on BG1

**Comment by hinyan17** [comment]
Working on B16

**Comment by Rakesh-Ranga-Buram** [comment]
My team will be working on BG2,BG5 and C2

**Comment by zhenga5** [comment]
I want to work on BG7

**Comment by huss** [comment]
Tests BG8, BG9, BG16, C3, C15, C16, CG2, CG15 & CG16 are all reserved for other developers.

**Comment by shankarp05** [comment]
Will work on BG10

**Comment by aravindb24** [comment]
Will work on BG11

**Comment by Emily-D-2003** [comment]
Added test BG14: https://github.com/OpenEnergyDashboard/OED/pull/1352

**Comment by nicholasyfu1** [comment]
Working on BG12!

**Comment by nicholasyfu1** [comment]
BG12 PR: https://github.com/OpenEnergyDashboard/OED/pull/1353

**Comment by omarraf** [comment]
Working on BG9

**Comment by EBoyd3443** [comment]
Working on BG8

**Comment by Chocopepero** [comment]
Working on C3

**Comment by tien-han** [comment]
My team is working on C16.

@sagemar & @cmatthews444

**Comment by BrendenHaskins** [comment]
Working on CG2

**Comment by brandonviorato** [comment]
We are working on CG15

@vladoleksiyenko @MattMiss @Gene7Him

**Comment by cervantes-yadira** [comment]
Working on BG13

**Comment by katefike** [comment]
My team is working on CG3.

**Comment by ghost** [comment]
My team is working on CG16

**Comment by developersbm** [comment]
My team will be working on C15

**Comment by katefike** [comment]
> Working on BG8

ok this is our assignment @Gballreich @dalejk88 @DPJProgramming

**Comment by huss** [comment]
> > Working on BG8
> 
> ok this is our assignment @Gballreich @dalejk88 @DPJProgramming

It seems two groups may be working on the same issue. If so, this needs to be resolved. Shifting one to another test case would fix this up. Please comment/discuss to resolve. Thanks.

**Comment by katefike** [comment]
> > > Working on BG8
> > 
> > 
> > ok this is our assignment @Gballreich @dalejk88 @DPJProgramming
> 
> It seems two groups may be working on the same issue. If so, this needs to be resolved. Shifting one to another test case would fix this up. Please comment/discuss to resolve. Thanks.

@huss - Sorry for the confusion, there are not two groups working on the same issue. We are all one group: myself @Gballreich @dalejk88 @DPJProgramming and @EBoyd3443 (who initially claimed BG8)

**Comment by huss** [comment]
> > > > Working on BG8
> > > 
> > > 
> > > ok this is our assignment @Gballreich @dalejk88 @DPJProgramming
> > 
> > 
> > It seems two groups may be working on the same issue. If so, this needs to be resolved. Shifting one to another test case would fix this up. Please comment/discuss to resolve. Thanks.
> 
> @huss - Sorry for the confusion, there are not two groups working on the same issue. We are all one group: myself @Gballreich @dalejk88 @DPJProgramming and @EBoyd3443 (who initially claimed BG8)

Sounds good and thanks to the team for working on this.

**Comment by LGH831** [comment]
My team is working on C14.

**Comment by huss** [comment]
@benkhong27 You should look at the Developer channel on the OED Discord server for a recent discussion on an SQL bug. I believe it applies to CG16 as well. The advice for the team doing C16 applies to you as well and I think your test will fail until the patch is applied. I think the correct values for CG16 are [1401748.41100155, 1460549.57015091] and I think you will see a value 100 times smaller until the patch is applied. Please feel free to contact me for further information/help. Sorry about this.

**Comment by huss** [comment]
> @benkhong27 You should look at the Developer channel on the OED Discord server for a recent discussion on an SQL bug. I believe it applies to CG16 as well. The advice for the team doing C16 applies to you as well and I think your test will fail until the patch is applied. I think the correct values for CG16 are [1401748.41100155, 1460549.57015091] and I think you will see a value 100 times smaller until the patch is applied. Please feel free to contact me for further information/help. Sorry about this.

@benkhong27 As discussed on the developer Discord channel, I just updated the expected values for CG16. Please use these values and let me know if you have any issue. Sorry about this.

**Comment by huss** [comment]
Tests BG7, C4, C5, C6, C8, CG3, CG4, CG5 & CG6 are reserved for other developers.

**Comment by miaescobar1** [comment]
I'm working on this!

**Comment by huss** [comment]
> I'm working on this!

@miaescobar1 This is an issue that has lots of tests where one person/team normally does one test. Given this, is there a specific test you are working on?

**Comment by miaescobar1** [comment]
My team and I are going to work on Test Case C6. Thanks for your prompt check-in!

**Comment by qopci** [comment]
Hello,

My team and I will be working on the Test Case C8. Thank you for the opportunity!

**Comment by alstondsouza1** [comment]
Hello there!

My team and I will be working on Test Case C5.
Thank you for this opportunity.

**Comment by abdelrhamannianaa** [comment]
My team and I will be working on test case CG4, thank you!

**Comment by thuynh-9084** [comment]
Hello, my team and I are going to be working on Test Case C4. Thank you!

**Comment by Unknown31414652** [comment]
Hi, my team and I will be working on test case CG3. Thank you!

**Comment by VanishedAIR** [comment]
Hello, my team and I will be working on CG6. Thank you!

**Comment by mRosenthal937** [comment]
My team will work on CG5. Thank you!

**Comment by BoLiuV5** [comment]
My team will work on CG5. Thanks.

**Comment by huss** [comment]
> My team will work on CG5. Thanks.

@bo2020bo I'm hoping you are working with @mRosenthal937 as they previously indicated they were working with a team on test case CG5. Please let me know if this is not the case.

**Comment by ChikaStarks** [comment]
Added test case CG4 (PR #1407)

**Comment by huss** [comment]
C9, C10 & C11 are reserved for other developers.

**Comment by huss** [comment]
C9, C10 & C11 are reserved for other developers.

**Comment by huss** [comment]
> Working on BG13

@cervantes-yadira I don't recall any work on this nor is there a pull request. Is this effort being done or should this test case be reassigned?

**Comment by huss** [comment]
CG8, C12 & C13 are reserved for other developers.

**Comment by huss** [comment]
> > Working on BG13
> 
> [@cervantes-yadira](https://github.com/cervantes-yadira) I don't recall any work on this nor is there a pull request. Is this effort being done or should this test case be reassigned?

To support other developers, I am un-reserving this issue for you and using it for someone else. If you need an issue then reach out to the project.

**Comment by huss** [comment]
Test cases C9, C10, C11, C12, C13 & C14 are reserved for other developers.

Note that at this time there are no other test cases available to do. OED may add more in the future.

**Comment by shaikhibrahim2000** [comment]
I am working on test BG13

**Comment by knakatasf** [comment]
I am working on the test case C13.

**Comment by huss** [comment]
> I am working on the test case 13.

@knakatasf Thanks for working on a test case. Just to clarify/be certain, do you mean C13? Thanks.

**Comment by yechanlin** [comment]
I am working on test case C12.

**Comment by knakatasf** [comment]
> > I am working on the test case 13.
> 
> [@knakatasf](https://github.com/knakatasf) Thanks for working on a test case. Just to clarify/be certain, do you mean C13? Thanks.

Yes. Sorry, I corrected.

**Comment by KViruz2750** [comment]
I am working on test case CG8.

**Comment by Phinhas214** [comment]
I'm working on this.

**Comment by kwohyuno** [comment]
C11 : Hyun Oh Kwon/Mon Raphael Fernandez/Abas Mohamed (students), and Catherine Chen (mentors)

**Comment by aizenbaidya** [comment]
~~**C14**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)~~

~~Thank you.~~

Whoops, it looks like test case C14 has already been completed. Sorry! 

https://github.com/OpenEnergyDashboard/OED/pull/1376

**Comment by huss** [comment]
> I'm working on this.

@Phinhas214 Thanks for working on a test case. Could you clarify which specific test you are working on? Thanks.

**Comment by Phinhas214** [comment]
> > I'm working on this.
> 
> [@Phinhas214](https://github.com/Phinhas214) Thanks for working on a test case. Could you clarify which specific test you are working on? Thanks.

I'm working on C9 test case.

**Comment by anthonytheng** [comment]
C10: Caden Carpenter, Helen Ton, Anthony Theng (Students)
Sunandan Barman (Mentor)

**Comment by aizenbaidya** [comment]
**B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)

Thank you!

**Comment by huss** [comment]
> **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)

Sounds good. I'm sorry you were previously given an assigned issue and appreciate your continued work on OED test cases.

**Comment by huss** [comment]
> > **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)
> 
> Sounds good. I'm sorry you were previously given an assigned issue and appreciate your continued work on OED test cases.

@aizenbaidya This is a heads up that I just added a number of placeholder comments for the new OED tests. B17 is in ``src/server/test/web/readingsBarMeterFlow.js`` and this change might cause a merge conflict with your work. The new comment for B17 will go away and your test code will go there (probably where you would put it anyway). If you get a conflict and want me to resolve it then I can do that once the pull request is in (or you can carefully do it yourself). Sorry for another complication and don't hesitate to ask for any information/help.

**Comment by aizenbaidya** [comment]
> [@aizenbaidya](https://github.com/aizenbaidya) This is a heads up that I just added a number of placeholder comments for the new OED tests. B17 is in `src/server/test/web/readingsBarMeterFlow.js` and this change might cause a merge conflict with your work. The new comment for B17 will go away and your test code will go there (probably where you would put it anyway). If you get a conflict and want me to resolve it then I can do that once the pull request is in (or you can carefully do it yourself). Sorry for another complication and don't hesitate to ask for any information/help.

No worries! Thank you for your continued support in helping our team contribute to open source!

**Comment by albertfast** [comment]
> **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)
> 
> Thank you!

B17: I’m working on the readingsBarMeterFlow.js test file

**Comment by huss** [comment]
> > **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)
> > Thank you!
> 
> B17: I’m working on the readingsBarMeterFlow.js test file

@albertfast Could you clarify what you are doing. Are you also working on B17 or are you just working in the same file (which is normal/fine)? Thanks.

**Comment by albertfast** [comment]
> > > **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)
> > > Thank you!
> > 
> > B17: I’m working on the readingsBarMeterFlow.js test file
> 
> @albertfast Could you clarify what you are doing. Are you also working on B17 or are you just working in the same file (which is normal/fine)? Thanks.

@huss I’ve written the B17 test in the readingsBarMeterFlow.js file, and it works flawlessly with the related CSV files when running independently without other dependencies. However, when connecting to PostgreSQL via the Docker container, I encounter issues that I’m actively working on resolving to fully run the test.

**Comment by huss** [comment]
> > > > **B17**: Ahmet Sahiner, Khang Nguyen, Aizen Baidya (students), and Sharanya Srivastava (mentor)
> > > > Thank you!
> > > 
> > > 
> > > B17: I’m working on the readingsBarMeterFlow.js test file
> > 
> > 
> > [@albertfast](https://github.com/albertfast) Could you clarify what you are doing. Are you also working on B17 or are you just working in the same file (which is normal/fine)? Thanks.
> 
> [@huss](https://github.com/huss) I’ve written the B17 test in the readingsBarMeterFlow.js file, and it works flawlessly with the related CSV files when running independently without other dependencies. However, when connecting to PostgreSQL via the Docker container, I encounter issues that I’m actively working on resolving to fully run the test.

@albertfast Ok. I was not sure if you were part of the team that said they were working on B17. I just looked at your GitHub ID and it seems to be one of the names so all seems fine.

As for the PostgreSQL issue. I'm unsure why you are connecting to that Docker container. When running tests via npm (the standard way) one does it in the OED web Docker container (maybe that is what you meant). Note in that Docker container you cannot access the database (PostgreSQL) - that must be done via the OED database Docker container. Let me know if this resolves the issue or if I am missing something.

**Comment by Gabby-Moon** [comment]
Me and my team are working on C17

**Comment by Dani-DEV28** [comment]
I'm working on B19 with my team

**Comment by AAshGray** [comment]
B22: Matthew Larsen/Ash Gray/Xalli Bell/Niko Butenko (students), and Anwell Wang (mentors)

**Comment by Bolshialex** [comment]
B21: Kamran Totakhil/Abdi Ahmed/Alex Bolshakov/Emily Menken (students), and Mohit Menghnani (mentors)

**Comment by dangrabo** [comment]
B20: Nur Mohamed, Daniel Grabowski, Beza Legeesse, Hunter Yates, (Students) and Parul Gupta (Mentor).

**Comment by xavierb117** [comment]
C18: Xavier Bradley, Tim Williams, Rudolf Sirbu, Shawn Nguru, and Rajesh Daruvuri (Mentor)

**Comment by RJRico17** [comment]
B18: RJ Fredrick Rico, Justin Saephan, Shams Al Tuwaijari, Grace Gong (Mentor)

**Comment by KySuongLam** [comment]
Hello,
I'm working on the test case **C20**: "_28 day shift end 2022-10-31 17:12:34 (partial hour) for 15 minute reading intervals and quantity units & kWh as kWh_" with my teammate **Selena Sat**.

**Comment by braedonbillingsley** [comment]
My team and I are working on test case C19. Jinan Almaliky, Domingo Martin, Braedon Billingsley (students), and David Asamu (mentor)

**Comment by chaitanyarahalkar** [comment]
Hi, 

My team and I are working on test case LR9. Conrad Nark, Rebecca Riffle, Hannah Angel (students) and Chaitanya Rahalkar (Mentor)

**Comment by S-in-PNW** [comment]
Hello, my team is working on test case L26. Thank you

**Comment by maigbao** [comment]
Hi, I am working on LR25

**Comment by MattMiss** [comment]
Hi, my team is working on LG26

**Comment by lukeckk** [comment]
Hi there, my team is working on L27

**Comment by mudra-senjaliya** [comment]
Hi there, my team is working on LR5

**Comment by ximixu** [comment]
Hi, my team is working on LR6

**Comment by lukeckk** [comment]
HI, I will be working on LR2

**Comment by TVW96** [comment]
Hello, I'm working on LR4.

**Comment by ximixu** [comment]
> Hi, my team is working on LR6

My team is no longer working on LR6, so it should be open.

**Comment by poojenie** [comment]
Hi, my team will be working on LR3

**Comment by tinisha-davis** [comment]
Hi coding friends, I'm working on LR6.

**Comment by HienXuanPham** [comment]
Hi, my team from CodeDay will work on LR3 and LR7

**Comment by huss** [comment]
> Hi, my team from CodeDay will work on LR3 and LR7

@HienXuanPham Thanks for letting us know. Normally a single team only does one of these issues so there are plenty to go around. Did CodeDay assign two to your single team or are you talking about multiple teams?

**Comment by HienXuanPham** [comment]
> > Hi, my team from CodeDay will work on LR3 and LR7
> 
> [@HienXuanPham](https://github.com/HienXuanPham) Thanks for letting us know. Normally a single team only does one of these issues so there are plenty to go around. Did CodeDay assign two to your single team or are you talking about multiple teams?

@huss CodeDay assigned these two to my team. They mentioned in the email this: "We will work on fixing the following test cases in the OED project: Test Case LR3 in issue https://github.com/OpenEnergyDashboard/OED/issues/962 Test Case LR7 in issue https://github.com/OpenEnergyDashboard/OED/issues/962"

**Comment by JazzJE** [comment]
Hi, my team is working on test case BG21.

**Comment by fluffylenny** [comment]
Hello, my team is working on test case LR1.

**Comment by Kaiying-Yu** [comment]
Hi, my team is working on test case CG17.

**Comment by fluffylenny** [comment]
Hello, My team is working on test case CG20.

**Comment by princetonn** [comment]
Hello, my team is working on test case CG9.

**Comment by stockingstocker** [comment]
Hi, my team from Codeday will be working on test case LR14.

**Comment by vanessagarciacruz** [comment]
Hello, my team is working on test case LR15.

**Comment by Adam-Alhakam** [comment]
Hi, my team from Codeday will be working on test case LR14.

**Comment by huss** [comment]
> Hi, my team from Codeday will be working on test case LR14.

@Adam-Alhakam Last week @stockingstocker indicated they were working on LR14. Are you working together? If not, I want to be sure you are not working on the same test case.

**Comment by stockingstocker** [comment]
@huss very sorry for the miscommunication, we are indeed working together on the same test case.

**Comment by huss** [comment]
> [@huss](https://github.com/huss) very sorry for the miscommunication, we are indeed working together on the same test case.

No problem. I'm glad it is not an issue. Let me know if you need anything.

**Comment by gokulkrishnakotta** [comment]
My team is working on the test case CG10.

**Comment by Ryan-z-Feng-ccsf** [comment]
Our team is working on BG17.

**Comment by aaronalaman** [comment]
My team is working on BG18.

**Comment by AndrewKam123** [comment]
My team is working on LR26.

**Comment by liliann19** [comment]
My team is working on test BG19.

**Comment by ragib-nehal** [comment]
My team is working on BG20.

**Comment by leslyVillanueva1020** [comment]
My team is working on BG21

**Comment by huss** [comment]
> My team is working on BG21

@leslyVillanueva1020 BG21 is already completed and in the development branch. Any chance you meant another test case?

**Comment by leslyVillanueva1020** [comment]
@huss Hi, On my team's intro email and the CodeDay dashboard it says we will be covering BG21.

**Comment by huss** [comment]
> [@huss](https://github.com/huss) Hi, On my team's intro email and the CodeDay dashboard it says we will be covering BG21.

@leslyVillanueva1020 Okay. You can check me but I think if you search the code base for BG21 it will be done with code. Unless you think I have this wrong, can you reach out to CodeDay to say I think the issue is done and should you be working on a different one?

**Comment by trivankute** [comment]
My team is working on BG22.

**Comment by 77amyfly** [comment]
My team is working on Test LR10

**Comment by Nebulaics** [comment]
My team is working on Test Case CG18

**Comment by Khadijasaho** [comment]
My team is working on Test Case CG11

**Comment by johnny603** [comment]
I'm working on this

**Comment by huss** [comment]
> I'm working on this

Could you clarify what you mean? This issue is a general one to track work on the reading test cases. Normally it's work done by new developers to OED and the case being worked on is provided by OED or it's partner organizations. Let me know if I can help.

**Comment by johnny603** [comment]
> > I'm working on this
> 
> Could you clarify what you mean? This issue is a general one to track work on the reading test cases. Normally it's work done by new developers to OED and the case being worked on is provided by OED or it's partner organizations. Let me know if I can help.

I was assigned to work on this with a teammate for Codeday's Microinternship for May along with a mentor, would you like us to work on another issue? Or is this one fine?

**Comment by huss** [comment]
> I was assigned to work on this with a teammate for Codeday's Microinternship 

@johnny603 That is fine as CodeDay assigns these as a partner organizations. Could you comment on which test case you are supposed to do so everyone knows. Thanks.

**Comment by KarnbirKhera** [comment]
Hi, our team from CodeDay is working on Test Case LR26

**Comment by johnny603** [comment]
> > I was assigned to work on this with a teammate for Codeday's Microinternship
> 
> [@johnny603](https://github.com/johnny603) That is fine as CodeDay assigns these as a partner organizations. Could you comment on which test case you are supposed to do so everyone knows. Thanks.

Hi how do we choose what test cases are available? Thanks for the clarification

**Comment by johnny603** [comment]
> > I was assigned to work on this with a teammate for Codeday's Microinternship 
> 
> @johnny603 That is fine as CodeDay assigns these as a partner organizations. Could you comment on which test case you are supposed to do so everyone knows. Thanks.

Test case LR18

**Comment by AdrienSch** [comment]
Hello there, sorry for the late response! I was also assigned to work on this issue with my teammate for CodeDay's May Micro-internship along with our mentor. The Test Case we've been tasked with is Test Case LR15.

**Comment by Angel43v3r** [comment]
Hi I was part of a CodeDay team for April that worked on test case CG11, but I am doing another CodeDay for May and this time my team is working on test case for LR22.

**Comment by huss** [comment]
> Hi I was part of a CodeDay team for April that worked on test case CG11, but I am doing another CodeDay for May and this time my team is working on test case for LR22.

@Angel43v3r Normally someone only does one test case and then moves on to another type of issue.  If CodeDay assigned the test case to you then it is ok but you can also check if another issue might be better.  Let me know if you need anything.

**Comment by Angel43v3r** [comment]
@huss I already submitted CG11 https://github.com/OpenEnergyDashboard/OED/pull/1621, just waiting for the approval to merge it and yes Codeday assigned LR22 to my new team. I do have a question about LR22, I looked at the code yesterday and it looks like the file for it is missing. I see it in the OED GitHub DesignDocs/testing/readingsData/ folder, but it's missing in the development branch src/server/test/web/readingsData folder. Do I have to copy the file and transfer it manually?

**Comment by huss** [comment]
@Angel43v3r 

> I do have a question about LR22, I looked at the code yesterday and it looks like the file for it is missing. I see it in the OED GitHub DesignDocs/testing/readingsData/ folder, but it's missing in the development branch src/server/test/web/readingsData folder. Do I have to copy the file and transfer it manually? 

Per step 7 of the testing design doc, one needs to:

> In general, you need to copy the file from the design document testing/readingsData/ directory to the src/server/test/web/readingsData/ directory since the expected files are not included in the OED code until a test is written

I hope this clarifies this.  Please let me know if you need anything.

**Comment by sabinMas** [comment]
I'm working on this - Test Case LR19

**Comment by BushyMustache** [comment]
I'm working on this - Test Case LR11

**Comment by seadrahbe** [comment]
Working in a CodeDay team on test case CG19: 

[Test Case CG19] Testing readings #962

**Comment by huss** [comment]
> I'm working on this

@sabinMas That you for the comment. Could you clarify which test cast you are working on? Thanks.

**Comment by Angel43v3r** [comment]
@huss we would like to work on LR17 next, is that free to take?

**Comment by huss** [comment]
> [@huss](https://github.com/huss) we would like to work on LR17 next, is that free to take?

@Angel43v3r There are a limited number of test cases so OED reserves them for first-time contributors. While OED appreciates your interest in doing more work, you will not be able to do another test case. The fact that they must be assigned by someone else is reflected by the R on the status in the testing design document to show they are all reserved. You are welcome to look for another issue. Please let me know if you need anything.

**Comment by Angel43v3r** [comment]
@huss sounds good, we will look at the issues. Also we saw a minor error from a previous test case on readingsLineMeterRangeRaw.js. Could you let me know if there are any specific template or process I should follow to submit about this?

**Comment by huss** [comment]
> Also we saw a minor error from a previous test case on readingsLineMeterRangeRaw.js. Could you let me know if there are any specific template or process I should follow to submit about this?

@Angel43v3r The best way is to submit a PR with a proposed change. If you want to ask me about it before doing that then message me.

**Comment by AaronB22** [comment]
Working on CG12.

**Comment by hebert87** [comment]
Hi @huss, my team was assigned issue LR17 by Daniel Bachhuber from a CodeDay project, and we wanted to check if it’s still available for us to take. Thanks!

**Comment by huss** [comment]
> my team was assigned issue LR17 by Daniel Bachhuber from a CodeDay project, and we wanted to check if it’s still available for us to take

@hebert87 Yes it is. If CodeDay assigns it then it should be available as we coordinate what they give. Let me know if you need anything.
